src/server/services/wms/qgswmsrenderer.cpp

```diff
@@ -127,6 +127,11 @@ namespace QgsWms
       settings.setMapScale( mapSettings.scale() );
       settings.setMapUnitsPerPixel( mapSettings.mapUnitsPerPixel() );
     }
+    else
+    {
+      double defaultMapUnitsPerPixel = QgsServerProjectUtils::wmsDefaultMapUnitsPerMm( *mContext.project() ) / mContext.dotsPerMm();
+      settings.setMapUnitsPerPixel( defaultMapUnitsPerPixel );
+    }
```

Comment:
I think it should probably be set in case of a non empty `RULE` parameter too (in `getLegendGraphics(QgsLayerTreeModelLegendNode &nodeModel)` method).

Reply:
Oh yes. Thanks a lot for that.